websites/F/FurAffinity/presence.ts

```diff
@@ -1,134 +1,178 @@
+					?.textContent.trim()}'s journal`;
+				presenceData.largeImageKey = profilePic;
+			} else if (!journalName) presenceData.details = "Creates a Journal";
 			else {
-				presenceData.details = "Updates a Journal";
-				presenceData.state = `"${journalName}"`;
+				presenceData.details = "Updating a journal";
+				presenceData.state = journalName;
 			}
-		} else if (
-			document.location.pathname.includes("/msg/pms/") &&
-			showBrowsingNotes &&
-			showBrowsingAccount
-		) {
-			switch (document.location.hash) {
+			presenceData.buttons = [
+				{
+					label: "Read Journal",
+					url: href,
+				},
+			];
+			break;
+		}
+		case "msg": {
+			switch (hash) {
 				case "#message": {
-					presenceData.details = "Reads a note";
+					presenceData.details = "Reading a note";
 					presenceData.state = `"${
 						document.querySelector("#message h2").textContent
 					}"`;
 					break;
 				}
 				case "#MsgForm": {
-					presenceData.details = "Writes a note";
+					presenceData.details = "Writing a note";
 					break;
 				}
 			}
-		} else {
-			presenceData.details = null;
-			presenceData.state = null;
+			break;
+		}
+		case "view": {
+			presenceData.details = document.querySelector(
+				'[class="submission-title"]'
+			)?.textContent;
+			presenceData.state = document.querySelector(
+				'[class="submission-id-sub-container"] > a > strong'
+			)?.textContent;
+			presenceData.buttons = [
+				{
+					label: "View post",
+					url: href,
+				},
+			];
+			break;
+		}
+		default: {
+			if (document.querySelector("userpage-nav-header") && username) {
+				if (current === "home") {
+					presenceData.details = `Viewing ${username}'s profile`;
+					presenceData.buttons = [
+						{
+							label: strings.buttonViewProfile,
+							url: href,
+						},
+					];
+				} else {
+					presenceData.details = `Viewing ${username}'s
+					${current}`;
+				}
+				presenceData.largeImageKey = profilePic;
+			} else if (document.querySelector('[class="section-header"]')) {
+				presenceData.details = `Reading ${
+					document
+						.querySelector('[class="section-header"]')
+						?.querySelector("h2")?.textContent
+				}`;
+			} else presenceData.details = strings.browse;
 		}
 	}
-}
-setInterval(checkCurrentPage, 1000);
-presence.on("UpdateData", async () => {
-	showBrowsingArt = await presence.getSetting<boolean>("browse");
-	showBrowsingProfile = await presence.getSetting<boolean>("profile");
-	showCreateJournal = await presence.getSetting<boolean>("journal");
-	showBrowsingAccount = await presence.getSetting<boolean>("account");
-	showBrowsingNotes = await presence.getSetting<boolean>("notes");
-	showBrowsingSubmissions = await presence.getSetting<boolean>("submissions");
-	showBrowsingSearch = await presence.getSetting<boolean>("search");
-	showBrowsingCategory = await presence.getSetting<boolean>("category");
-	presenceData.startTimestamp = browsingTimestamp;
+	if (!search?.value && !presenceData.buttons && pathname !== "/") {
```

Comment:
I forgot that it returrs when searching
```suggestion
	if (!presenceData.buttons && pathname !== "/") {
```